### ISSUE TITLE
.proto enum variable named the same as enumerator causes compilation error

### ISSUE THREAD

**Issue by maregrabik** [square/wire#299]
We got proto file:

```
enum Gender {
   Gender_Male = 0;
   Gender_Female = 1;
}
message SanctionPerson {
   optional string FirstName = 1;
   optional Gender Gender = 2;
}

After compiling to java:
<b>Gender.java</b>

public enum <b>Gender</b>
    implements ProtoEnum {
  Gender_Male(0),
  Gender_Female(1);

  private final int value;

  Gender(int value) {
    this.value = value;
  }

  public int getValue() {
    return value;
  }
```

<b>SanctionPerson.java:</b>

```
public final class SanctionPerson extends Message {
  private static final long serialVersionUID = 0L;

  public static final String DEFAULT_FIRSTNAME = "";
 <b> public static final Gender DEFAULT_GENDER = Gender.Gender_Male;// Compile error Cannot make a static reference to the non-static field Gender</b>

  @ProtoField(tag = 1, type = STRING)
  public final String FirstName;

  @ProtoField(tag = 2, type = ENUM)
  public final <b>Gender Gender; //Field got the same name</b>

  public SanctionPerson(String FirstName, Gender Gender) {
    this.FirstName = FirstName;
    this.Gender = Gender;
  }
```

The problem is that 
 public static final Gender DEFAULT_GENDER = <b>Gender.Gender_Male</b> 
is referring to local variable instead Gender enum in Gender.java.

Or is there any possibility to force compiler to use for example camelCase for variables in .proto like 
<b>optional Gender Gender = 2;</b>?
Error from console: "non-static variable Gender cannot be referenced from a static context" in compile time.

**Comment by JakeWharton** [comment]
This naming convention violates the proto style guide: https://developers.google.com/protocol-buffers/docs/style. You are able to rename the field without affecting any clients, as an easy way to quickly eliminate the problem.

**Comment by maregrabik** [comment]
Thanks for quick response!
Unfortunately i cant change anything in base .proto , i have to work on given files. I am aware of that it`s braking conventions, but still prottobuff with it`s own compiler transform it to non error java class. Hoped to switch for wire`s one, but that situation block me out.

**Comment by mellster2012** [comment]
Facing the same issue. Wonder how hard a PR would be, doesn't seem like an overly complex problem to fix but it would help greatly for those that don't have control over the proto files given to them.

**Comment by mellster2012** [comment]
The above pull request worked for my project. All cases were simple field names clashing with their simple class/type names and this additional check will add the default package extension for field names (like it already does for colliding field names) in case of a clash.

**Comment by oldergod** [comment]
Fixed via #1252
